daml-foundations/daml-ghc/daml-stdlib-src/DA/Math.daml

```diff
@@ -0,0 +1,188 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+daml 1.2
+-- | Math - Utility Math functions for `Int` and `Decimal`
+module DA.Math
+( Powerable(..)
+, sin
+, cos
+, tan
+, pi
+) where
+
+import DA.List
+
+infixr 8 ^^
+infixr 8 **
+
+class (Multiplicative a, Fractional a) => Powerable a where
+  -- | Take an integer power of a number. Example: `2.0 ^^ 2 == 4.0`.
+  (^^) : a -> Int -> a
```

Comment:
So for reference, OCaml uses `(**)` for integer exponentiation and `(**.)` for floating point. Why though are there two operators in this class and not just `(**) : a -> a -> a`?

Reply:
Don't understand the question. You mean why have a class at all?

Reply:
On reflection, `(^^)` is traditionally spelled `pow`. I think that's the better name here. `(**)` I like (prior art as mentioned above).

Reply:
Ah, I see now that `(^)`, `(^^)` and `(**)` are "normal" in Haskell. Ok, question resolved for me.

Reply:
They are normal, but confuse me as an "expert" Haskeller. If you want to do better that's OK.